CMakeLists.txt

```diff
@@ -555,98 +557,22 @@ get_property(OUTPUT_FORMAT        GLOBAL PROPERTY PROPERTY_OUTPUT_FORMAT)
-  # that zephyr libraries follow a naming scheme as described in
-  # extensions.cmake:zephyr_library_get_current_dir_lib_name
-  #
-  # But code from test/ and lib/ that is placed in the "zephyr"
-  # library (with zephyr_sources()) will not be in a library that is
-  # prefixed with lib__ or test__ and will end up in the wrong address
-  # space.
-  set(application_space_dirs
-    lib
-    tests
-    )
-  foreach(f ${kernel_object_file_list})
-    foreach(app_dir ${application_space_dirs})
-      if(${f} MATCHES "^${app_dir}__") # Begins with ${app_dir}__, e.g. lib__libc
-        list(
-          REMOVE_ITEM
-          kernel_object_file_list
-          ${f}
-          )
-      endif()
-    endforeach()
-  endforeach()
-
-  # Create a list ks, with relative paths to kernel space libs.
-  foreach(f ${kernel_object_file_list})
-    get_target_property(target_name       ${f} NAME)
-    get_target_property(target_binary_dir ${f} BINARY_DIR)
-
-    string(REPLACE
-      ${PROJECT_BINARY_DIR}
-      ""
-      fixed_path
-      ${target_binary_dir}
-      )
-
-    # Append / if not empty
-    if(fixed_path)
-      set(fixed_path "${fixed_path}/")
-    endif()
-
-    # Cut off leading / if present
-    if(fixed_path MATCHES "^/.+")
-      string(SUBSTRING ${fixed_path} 1 -1 fixed_path)
-    endif()
-
-    set(fixed_path "${fixed_path}lib${target_name}.a")
-
-    if(CMAKE_GENERATOR STREQUAL "Ninja")
-      # Ninja invokes the linker from the root of the build directory
-      # (APPLICATION_BINARY_DIR) instead of from the build/zephyr
-      # directory (PROJECT_BINARY_DIR). So for linker-defs.h to get
-      # the correct path we need to prefix with zephyr/.
-      set(fixed_path "zephyr/${fixed_path}")
-    endif()
-
-    list(APPEND ks ${fixed_path})
-  endforeach()
-
-  # We are done constructing kernel_object_file_list, now we inject this
-  # information into the linker script through -D's
-  list(LENGTH kernel_object_file_list NUM_KERNEL_OBJECT_FILES)
-  list(APPEND LINKER_SCRIPT_DEFINES -DNUM_KERNEL_OBJECT_FILES=${NUM_KERNEL_OBJECT_FILES})
-  set(i 0)
-  foreach(f ${ks})
-    list(APPEND LINKER_SCRIPT_DEFINES -DKERNEL_OBJECT_FILE_${i}=${f})
-    math(EXPR i "${i}+1")
-  endforeach()
-endif() # CONFIG_APPLICATION_MEMORY
+if(CONFIG_APPLICATION_MEMORY)
+  set(KERNEL_OBJECTS_H ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/kernel_objects.h)
+  set(IMACROS_KERNEL_OBJECTS_H -imacros ${KERNEL_OBJECTS_H})
+
+  add_custom_command(OUTPUT ${KERNEL_OBJECTS_H}
+    COMMAND ${PYTHON_EXECUTABLE} -c
+       \"list='$<TARGET_PROPERTY:memory_space_kernel,COMPILE_DEFINITIONS>'.split('\;') \;
+         s=('\#define NUM_KERNEL_OBJECT_FILES %s\\n' % len(list)) \;
+         s+=''.join('\#define KERNEL_OBJECT_FILE_%s %s\\n' % (i,item) for i, item in enumerate(list)) \;
+         f=open('${KERNEL_OBJECTS_H}','w') \;
+         f.write(s) \;
+         f.close() \"
+         COMMENT "Populating ${KERNEL_OBJECTS_H} with kernel space symbols"
```

Comment:
'kernel space symbols' implies the header file is encoded with symbols from the symbol table that are in kernel space. But in fact it is populated with:

```
#define NUM_KERNEL_OBJECT_FILES 4
#define KERNEL_OBJECT_FILE_0 zephyr/arch/libarch_arm.a
#define KERNEL_OBJECT_FILE_1 zephyr/subsys/bluetooth/libsubsys_bluetooth.a
#define KERNEL_OBJECT_FILE_2 zephyr/subsys/net/libsubsys_net.a
#define KERNEL_OBJECT_FILE_3 zephyr/kernel/libkernel.a

```

Reply:
will rephrase comment

Reply:
Done